openml/extensions/sklearn/extension.py

```diff
@@ -307,11 +307,16 @@ def _deserialize_sklearn(
                     rval = self._deserialize_rv_frozen(value)
                 elif serialized_type == "function":
                     rval = self._deserialize_function(value)
-                elif serialized_type == "component_reference":
+                elif serialized_type in ("composition_step_constant", "component_reference"):
+                    if serialized_type == "composition_step_constant":
+                        value = self._deserialize_parameter_step_constants(value)
+                    elif serialized_type == "component_reference":
+                        value = self._deserialize_sklearn(
+                            value, recursion_depth=depth_pp, strict_version=strict_version
+                        )
+                    else:
+                        raise NotImplementedError(serialized_type)
                     assert components is not None  # Necessary for mypy
```

Comment:
Not your code but any idea why the type hint is `Optional[dict]` and the doc-string has type `dict`? There is no documentation of what this function argument actually is.

Reply:
> Not your code but any idea why the type hint is Optional[dict] and the doc-string has type dict?

That's a mistake. It should be optional in the docstring, too.

Reply:
> There is no documentation of what this function argument actually is.

Will add this.